examples/mediatek/models/llm_models/modeling_common.py

```diff
@@ -0,0 +1,497 @@
+                    lm_head_weight_key = 'embed_tokens.weight'
+                    lm_head_bias_key = 'embed_tokens.bias'
+                else:
+                    lm_head_weight_key = 'lm_head.weight'
+                    lm_head_bias_key = 'lm_head.bias'
+                temp_state_dict = {**temp_state_dict,
+                    **{
+                        'lm_head.weight':                                                                state_dict.pop(lm_head_weight_key),
+                        'norm.weight':                                                                   state_dict.pop(f'{prefix}norm.weight').to(torch.float32),
+                        'lm_head.bias':                                                                  state_dict.pop(lm_head_bias_key, torch.zeros(self.config.vocab_size, dtype=self.dtype))
+                    }
+                }
+                if self.config.norm == "LayerNorm":
+                    temp_state_dict['norm.bias'] = state_dict.pop(f'{prefix}norm.bias', torch.zeros(self.config.hidden_size, dtype=self.dtype)).to(torch.float32)
+
+        print(f"Loading weights for chunk {self.chunk_idx}")
+        if temp_state_dict.keys() != self.state_dict().keys():
+            temp_state_dict_only_keys = [x for x in temp_state_dict.keys() if x not in self.state_dict().keys()]
+            model_only_keys = [x for x in self.state_dict().keys() if x not in temp_state_dict.keys()]
+            raise RuntimeError(f"model state dict keys don't match with state_dict to load into model.\nModel only keys:{model_only_keys}\nstate_dict only keys:{temp_state_dict_only_keys}")
+        self.load_state_dict(temp_state_dict)
+        for i in range(self.num_blocks):
+            self.layers[i].to(self.device_list[i])
+        if self.config.use_stable_embedding and self.chunk_idx == 0:
+            self.embed_layer_norm.to(self.device_list[0])
+        if self.include_tail:
+            self.norm.to(self.device_list[-1])
+            self.lm_head.to(self.device_list[-1])
+        self.eval()
+
+        return self
+
+    def get_example_inputs(
+        self,
+        num_token: int = 128,
+        cache_size: int = 512,
+        get_dym_shape = False
+    ):
+        head_dim = int(self.config.hidden_size/self.config.num_attention_heads)
+        example_inputs = (
+            torch.randn(1, num_token, self.config.hidden_size, device='cpu', dtype=torch.float32),
+            torch.randn(1, 1, num_token, cache_size+num_token, device='cpu', dtype=torch.float32),
+            torch.randn(1, 2, num_token, head_dim, device='cpu', dtype=torch.float32),
+            *[torch.randn(1, self.config.num_key_value_heads, cache_size, head_dim, device='cpu', dtype=torch.float32) for _ in range(2*self.num_blocks)]
+        )
+        # Note: Assume cache size fixed shape as torch dynamic shape cannot handle dim 3 being
+        # combination of 2 dynamic dims
+        if get_dym_shape:
```

Comment:
Are we using dynamic shape?

Reply:
Dynamic Shape is more for the case where users would like to calibrate on their own dataset. 

With dynamic shape, users can avoid padding each input to the length. 

Also, it is only used during the `capture_pre_autograd_graph`  part of the export flow. 

Subsequently the shape is fixed during the `torch.export.export` part